packages/network/src/connection/WebRtcEndpoint.ts

```diff
@@ -231,20 +227,108 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
+                this.logger.warn(e)
+            }
+
+            this.connections[peerId] = connection
+        }
+        else if (this.connections[peerId] && this.connections[peerId].getConnectionId() !== 'none') {
+            const conn = this.connections[peerId]
+            const lastState = conn.getLastState()
+            this.logger.trace('onRtcOffer: answering reconnect attempt; there is already a connection for %s. state: %s', 
+                NameDirectory.getName(peerId), lastState)
+            
+            connection = this.createConnection(peerId, routerId, false)
+            if (conn.getDeferredConnectionAttempt()) {
+                connection.setDeferredConnectionAttempt(conn.stealDeferredConnectionAttempt())
+            }
+            
+            delete this.connections[peerId]
+            conn.close()
+
+            try {
+                connection.connect()
+            }
+            catch(e) {
+                this.logger.warn(e)
+            }
+            this.connections[peerId] = connection
+        }
+        else {
+            connection = this.connections[peerId]
+        }
+       
+        connection.setPeerInfo(PeerInfo.fromObject(originatorInfo))
+        connection.setConnectionId(connectionId)
+
+        connection.setRemoteDescription(description, 'offer' as DescriptionType.Offer)
+    }
+
+    async connect(
+        targetPeerId: string,
+        routerId: string,
+        trackerInstructed = true
+    ): Promise<string> {
+        const offering = this.peerInfo.peerId < targetPeerId
+        const peerType = offering ? 'offerer' : 'answerer'
+
+        // Prevent new connections from being opened when WebRtcEndpoint has been closed
+        if (this.stopped) {
+            return Promise.reject(new WebRtcError('WebRtcEndpoint has been stopped'))
+        }
+
+        if (this.connections[targetPeerId]) {
+            const connection = this.connections[targetPeerId]
+            const lastState = connection.getLastState()
+            this.logger.trace('%s has already connection for %s. state: %s', peerType, NameDirectory.getName(targetPeerId), lastState)
+            if (lastState === 'connected') {
+                return Promise.resolve(targetPeerId)
+            }
+
+            if (connection.getDeferredConnectionAttempt()) {
+                return (connection.getDeferredConnectionAttempt()?.connectionAttemptPromise as Promise<string>)
+            }
+            else {
+                throw new Error(`disconnected ${connection.getPeerId()}`)
+            }
+        }
+        
+        const connection = this.createConnection(targetPeerId, routerId, offering)
+        connection.createDeferredConnectionAttempt()
+
+        if (offering) {
+            connection.setConnectionId(uuidv4())
+        }
 
         this.connections[targetPeerId] = connection
         connection.connect()
-        if (!isOffering) {
+        
+        if (!trackerInstructed && !offering) {
```

Comment:
Added a comment that may be useful for a new reader of the class:
```suggestion
        // If we are non-offerer and this connection was not instructed by the tracker, we need
        // to let the offering side know about it so it can send us the initial offer message.
        if (!trackerInstructed && !offering) {
```

Reply:
Added the comment and also renamed the method to call to sendRtcConnect() to be consistent